rust/main/agents/relayer/src/msg/metadata/base_builder.rs

```diff
@@ -287,4 +273,39 @@ impl BaseMetadataBuilder {
         fetch_storage_locations_helper(validators, &self.cache, &*self.origin_validator_announce)
             .await
     }
+
+    async fn build_and_validate(
+        &self,
+        config: &CheckpointSyncerConf,
+        validator: &H256,
+        latest_index_gauge: Option<IntGauge>,
+    ) -> Result<Option<Box<dyn CheckpointSyncer>>, CheckpointSyncerBuildError> {
+        match config.build_and_validate(latest_index_gauge).await {
+            Ok(checkpoint_syncer) => {
+                return Ok(Some(checkpoint_syncer));
+            }
+            Err(CheckpointSyncerBuildError::ReorgEvent(reorg_event)) => {
+                if self.ignore_reorg_reports {
+                    warn!(
+                        ?reorg_event,
+                        "Ignoring reorg event for checkpoint syncer build"
+                    );
+                    return Ok(None);
```

Comment:
all of the code in this function except for this snippet is pre-existing code that was moved out of `build_checkpoint_syncer` above